File: app/src/main/java/com/example/javarushproject/model/Dog.java
package com.example.javarushproject.model;

public class Dog extends Animal{
    public Dog(int height, int weight, String name) {
        super(height, weight, name);
    }

    @Override
    public String toString() {
        return "Dog{" +
                "height=" + super.getHeight() +
                ", weight=" + super.getWeight() +
                ", name=" + super.getName() +
                '}';
    }

}

File: app/src/main/java/com/example/javarushproject/model/Dog.kt
package com.example.javarushproject.model

class Dog(height: Int, weight: Int, name: String?) : Animal(height, weight, name!!) {
    override fun toString(): String {
        return "Dog{" +
                "height=" + super.height +
                ", weight=" + super.weight +
                ", name=" + super.name +
                '}'
    }
}